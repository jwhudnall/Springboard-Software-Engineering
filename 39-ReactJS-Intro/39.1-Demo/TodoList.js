const ToDoList = (props) => {
  return (
    <div>
      <h4>Todo List:</h4>
      <ul>
        {props.todos.map((i) => (
          <li>
            <input type='checkbox' />
            {i}
          </li>
        ))}
      </ul>
    </div>
  );
};
