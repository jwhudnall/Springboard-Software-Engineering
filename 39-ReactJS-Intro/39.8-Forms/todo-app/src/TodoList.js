import React, { useState } from "react";
import { v4 as uuidv4 } from "uuid";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
  const INITAL_STATE = [
    { task: "Walk the dog.", id: uuidv4() },
    { task: "Take out the Trash.", id: uuidv4() },
    { task: "Study.", id: uuidv4() }
  ];
  const [todos, setTodos] = useState(INITAL_STATE);
  const deleteTodo = (id) => {
    setTodos(
      todos.filter((t) => {
        return t.id !== id;
      })
    );
  };
  const addTodo = (newItem) => {
    setTodos((todos) => [...todos, { ...newItem }]);
  };

  return (
    <>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map((t) => (
          <Todo key={t.id} task={t.task} deleteTodo={() => deleteTodo(t.id)} />
        ))}
      </ul>
    </>
  );
};

export default TodoList;
