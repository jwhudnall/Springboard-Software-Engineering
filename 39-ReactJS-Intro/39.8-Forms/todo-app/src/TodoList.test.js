import { render, screen } from "@testing-library/react";
import TodoList from "./TodoList";

test("renders", () => {
  render(<TodoList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
