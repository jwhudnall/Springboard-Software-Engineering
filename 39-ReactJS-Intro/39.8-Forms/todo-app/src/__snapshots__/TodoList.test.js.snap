// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task
    </label>
    <input
      id="task"
      name="task"
      required=""
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
  <ul>
    <li>
      Walk the dog.
      <button>
        X
      </button>
    </li>
    <li>
      Take out the Trash.
      <button>
        X
      </button>
    </li>
    <li>
      Study.
      <button>
        X
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
